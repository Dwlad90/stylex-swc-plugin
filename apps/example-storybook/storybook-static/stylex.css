
@layer priority1, priority2, priority3, priority4;
@layer priority1{
.padding-x1bg5miv{padding:12px 24px}
.padding-x1tamke2{padding:16px}
.padding-x1arfzav{padding:6px 12px}
.padding-x1ff1495{padding:8px 16px}
}
@layer priority2{
.borderColor-x7kbyhx{border-color:#ddd}
.borderColor-x159w2ql{border-color:#e0e0e0}
.borderColor-x9r1u3d{border-color:transparent}
.borderRadius-x1kogg8i{border-radius:6px}
.borderRadius-xur7f20{border-radius:8px}
.borderStyle-xng3xce{border-style:none}
.borderStyle-x1y0btm7{border-style:solid}
.borderWidth-xc342km{border-width:0}
.borderWidth-xmkeg23{border-width:1px}
.textDecoration-x1hl2dhg{text-decoration:none}
.transition-xurpv0g{transition:all.2s ease}
}
@layer priority3{
.alignItems-x6s0dn4{align-items:center}
.backgroundColor-x11jye9f{background-color:#06c}
.backgroundColor-x1596o1h{background-color:#dc3545}
.backgroundColor-x6cgz0f{background-color:#f5f5f5}
.backgroundColor-x12peec7{background-color:white}
.boxShadow-x1mh2kpm{box-shadow:0 2px 4px rgba(0,0,0,.1)}
.boxShadow-xz3rqps{box-shadow:0 4px 8px rgba(0,0,0,.15)}
.boxShadow-x1gnnqk1{box-shadow:none}
.color-x1525slw{color:#333}
.color-x33gn0b{color:#666}
.color-x1awj2ng{color:white}
.cursor-x1ypdohk{cursor:pointer}
.display-x3nfvp2{display:inline-flex}
.fontFamily-xmcqa5u{font-family:-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,sans-serif}
.fontSize-xfifm61{font-size:12px}
.fontSize-xif65rj{font-size:14px}
.fontSize-x1j61zf2{font-size:16px}
.fontSize-xosj86m{font-size:18px}
.fontWeight-xk50ysn{font-weight:500}
.fontWeight-x117nqv4{font-weight:bold}
.justifyContent-xl56j7k{justify-content:center}
.lineHeight-x1evy7pa{line-height:1.5}
.transform-xnn1q72{transform:translateY(0)}
.backgroundColor-xkdfp31:hover{background-color:#0052a3}
.backgroundColor-x37qbrc:hover{background-color:#c82333}
.backgroundColor-x1jmbrsh:hover{background-color:#e8e8e8}
.boxShadow-x1yrdse8:hover{box-shadow:0 4px 8px rgba(0,0,0,.12)}
.transform-x14ow2ja:hover{transform:translateY(-1px)}
.transform-x3kbueh:active{transform:translateY(0)}
}
@layer priority4{
.marginBottom-x1e56ztr{margin-bottom:8px}
.maxWidth-xw5ewwj{max-width:300px}
.minHeight-xe0p6wg{min-height:28px}
.minHeight-xu0aao5{min-height:36px}
.minHeight-x1gg8mnh{min-height:44px}
}