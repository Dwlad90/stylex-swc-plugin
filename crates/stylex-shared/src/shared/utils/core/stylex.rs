use crate::{
  shared::{
    enums::data_structures::fn_result::FnResult, utils::ast::convertors::string_to_expression,
  },
  transform::styleq::common::styleq,
};

use super::parse_nullable_style::ResolvedArg;

pub(crate) fn stylex(values: &Vec<ResolvedArg>) -> Option<FnResult> {
  let result = styleq(values);

  Some(FnResult::Stylex(string_to_expression(
    result.class_name.as_str(),
  )))
}
