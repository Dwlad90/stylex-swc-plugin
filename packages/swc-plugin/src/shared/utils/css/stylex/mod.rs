pub mod evaluate;
pub mod evaluate_stylex_create_arg;
pub(crate) mod flatten_raw_style_object;
pub(crate) mod parse_css;
pub(crate) mod split_css_value;
