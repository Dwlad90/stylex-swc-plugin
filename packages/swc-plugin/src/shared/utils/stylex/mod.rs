pub(crate) mod attrs;
pub(crate) mod define_vars_utils;
pub(crate) mod dev_class_name;
pub(crate) mod js_to_expr;
pub(crate) mod make_string_expression;
pub(crate) mod member_expression;
pub(crate) mod parse_nullable_style;
pub(crate) mod props;
pub(crate) mod stylex;
pub(crate) mod stylex_merge;
pub(crate) mod tests;
