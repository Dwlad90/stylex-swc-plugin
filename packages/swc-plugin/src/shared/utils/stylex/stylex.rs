use crate::{
  shared::{enums::FnResult, utils::common::string_to_expression},
  transform::styleq::styleq::styleq,
};

use super::parse_nullable_style::ResolvedArg;

pub(crate) fn stylex(values: &Vec<ResolvedArg>) -> Option<FnResult> {
  let result = styleq(values);

  let class_name = result.class_name.to_string();

  // let props: Vec<PropOrSpread> = vec![];

  // panic!("Implement stylex");

  // if let Some(value) =  {
  // vec![];
  // };
  //  else {
  //     vec![]
// };

  dbg!(&values, &class_name);

  Some(FnResult::Stylex(string_to_expression(class_name).unwrap()))

  // Some(FnResult::Stylex(
  //     object_expression_factory(vec![prop_or_spread_expression_creator(
  //         String::from("className"),
  //         string_to_expression(class_name).unwrap(),
  //     )])
  //     .unwrap(),
  // ))
}

pub(crate) fn gen_condition_permutations(count: usize) -> Vec<Vec<bool>> {
  let mut result = Vec::new();
  for i in 0..2u32.pow(count as u32) {
    let mut combination = Vec::new();
    for j in 0..count {
      combination.push(i & (1 << j) != 0);
    }
    result.push(combination);
  }
  result
}
