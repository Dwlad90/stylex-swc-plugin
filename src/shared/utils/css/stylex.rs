use std::collections::HashMap;

use indexmap::IndexMap;
use swc_core::ecma::ast::{Id, KeyValueProp, VarDeclarator};

use crate::{
    shared::{
        structures::{
            injectable_style::{InjectableStyle, InjectableStyleBase}, meta_data::MetaData, pre_rule::{CompiledResult, PreRule, PreRules}, stylex_options::StyleXOptions, stylex_state_options::StyleXStateOptions
        },
        utils::{css::flatten_raw_style_object, validators::validate_and_return_property},
    },
};

pub(crate) fn stylex_create(
    namespace_name: &str,
    property: &KeyValueProp,
    prefix: &str,
    declarations: &Vec<VarDeclarator>,
    var_dec_count_map: &mut HashMap<Id, i8>,
    options: &StyleXStateOptions,
) -> (
    IndexMap<String, IndexMap<String, Option<String>>>,
    IndexMap<String, InjectableStyle>,
) {
    validate_and_return_property(property);

    let mut resolved_namespaces: IndexMap<String, IndexMap<String, Option<String>>> =
        IndexMap::new();
    let mut injected_styles_map: IndexMap<String, InjectableStyle> = IndexMap::new();

    // let css_property = get_key_str(property);

    let mut pseudos = vec![];
    let mut at_rules = vec![];

    let flattened_namespace = flatten_raw_style_object(
        property,
        declarations,
        var_dec_count_map,
        &mut pseudos,
        &mut at_rules,
        options,
    );
    println!("!!!!__ flattened_namespace: {:#?}", flattened_namespace);

    let compiled_namespace_tuples = flattened_namespace
        .iter()
        .map(|(key, value)| match value {
            PreRules::PreRuleSet(rule_set) => (key.to_string(), rule_set.clone().compiled(prefix)),
            PreRules::StylesPreRule(styles_pre_rule) => {
                (key.to_string(), styles_pre_rule.clone().compiled(prefix))
            }
            PreRules::NullPreRule(rule_set) => (key.to_string(), rule_set.clone().compiled(prefix)),
        })
        .collect::<Vec<(String, CompiledResult)>>();

    println!(
        "!!!!__ compiled_namespace_tuples: {:#?}",
        compiled_namespace_tuples
    );

    let compiled_namespace = compiled_namespace_tuples
        .iter()
        .map(|(key, value)| {
            (
                key.to_string(),
                match value {
                    CompiledResult::ComputedStyles(styles) => {
                        CompiledResult::ComputedStyles(styles.clone())
                    }
                    CompiledResult::Null(null) => CompiledResult::Null(null.clone()),
                    _ => todo!("handle other cases"),
                },
            )
        })
        .collect::<IndexMap<String, CompiledResult>>();

    println!("!!!!__ compiled_namespace: {:#?}", compiled_namespace);

    let mut namespace_obj: IndexMap<String, Option<String>> = IndexMap::new();
    for key in compiled_namespace.keys() {
        let value = compiled_namespace.get(key).unwrap();

        // ...

        if let Some(included_styles) = value.as_included_style() {
            todo!("handle included style")
            // namespace_obj.insert(key.clone(), Some(included_styles));
        } else if let Some(styles) = value.as_computed_styles() {
            // let mut class_name_tuples = styles
            //     .iter()
            //     .map(|item| item)
            //     .collect::<Vec<ComputedStyle>>();

            let class_name_tuples = styles.clone();

            let class_name = &class_name_tuples
                .iter()
                .map(|computed_style| {
                    let class_name = computed_style.0.clone();

                    println!("!!!!__ class_name to join: {:#?}", class_name);

                    return class_name;
                })
                .collect::<Vec<String>>()
                .join(" ");

            namespace_obj.insert(key.clone(), Some(class_name.clone()));

            println!(
                "!!!!__ class_name_tuples: {:#?}, namespace_obj: {:#?}",
                class_name_tuples, namespace_obj
            );

            for item in &class_name_tuples {
                let class_name = item.0.clone();
                let injectable_styles = item.1.clone();
                if !injected_styles_map.contains_key(class_name.as_str()) {
                    injected_styles_map.insert(class_name.clone(), injectable_styles.clone());
                }
            }
        } else {
            println!("!!!!__ value: {:#?}", value);
            namespace_obj.insert(key.clone(), Option::None);
        }
    }

    resolved_namespaces.insert(namespace_name.to_string(), namespace_obj);
    (resolved_namespaces, injected_styles_map)
    // let values: Vec<String> = flattened_namespace.values().map(|v| v.get_value().to_string()).collect();

    // for vec in flattened_namespace.values() {
    //     for pre_rule in vec {
    //         values.push(pre_rule);
    //     }
    // }

    // let (css_style, class_name_hashed, rules) = convert_style_to_class_name(
    //     (&css_property, &values),
    //     &mut pseudos,
    //     &mut at_rules,
    //     prefix,
    // );

    // println!("!!!!__ rules: {:?}", rules);

    // MetaData {
    //     class_name: class_name_hashed.clone(),
    //     style: InjectableStyleBase {
    //         // priority: Option::None, //MetaData::set_priority(&css_property),
    //         rtl: None,
    //         ltr: format!(".{}{}", class_name_hashed, css_style),
    //     },
    //     priority: MetaData::set_priority(&css_property),
    // }
}

pub(crate) fn evaluate_style_x_create_arg(){

}