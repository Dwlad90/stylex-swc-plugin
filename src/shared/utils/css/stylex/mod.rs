pub mod evaluate;
pub mod evaluate_stylex_create_arg;
pub(crate) mod flatten_raw_style_object;
