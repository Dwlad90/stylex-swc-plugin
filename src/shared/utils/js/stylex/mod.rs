pub(crate) mod stylex_create;
pub(crate) mod stylex_define_vars;
pub(crate) mod stylex_first_that_works;
pub(crate) mod stylex_include;
pub(crate) mod stylex_keyframes;
