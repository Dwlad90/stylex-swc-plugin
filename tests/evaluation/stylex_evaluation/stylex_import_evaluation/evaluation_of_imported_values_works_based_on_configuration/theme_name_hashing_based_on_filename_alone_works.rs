use stylex_swc_plugin::{shared::structures::plugin_pass::PluginPass, ModuleTransformVisitor};
use swc_core::ecma::{
    parser::{Syntax, TsConfig},
    transforms::testing::test,
};

test!(
    Syntax::Typescript(TsConfig {
        tsx: true,
        ..Default::default()
    }),
    |tr| ModuleTransformVisitor::new_test_styles(
        tr.comments.clone(),
        PluginPass::default(),
        Option::None
    ),
    importing_file_with_stylex_suffix_works,
    r#"
    import stylex from 'stylex';
    // import { MyTheme } from 'otherFile.stylex';
    const styles = stylex.create({
      green: {
        color: "red",
      }
    });
    stylex(styles.green);
    "#
);
