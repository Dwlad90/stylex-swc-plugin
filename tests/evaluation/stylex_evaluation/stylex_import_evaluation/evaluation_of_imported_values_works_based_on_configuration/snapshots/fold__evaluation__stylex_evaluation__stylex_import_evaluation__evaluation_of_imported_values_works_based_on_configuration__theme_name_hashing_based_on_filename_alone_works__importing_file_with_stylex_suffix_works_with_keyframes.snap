---
source: tests/evaluation/stylex_evaluation/stylex_import_evaluation/evaluation_of_imported_values_works_based_on_configuration/theme_name_hashing_based_on_filename_alone_works.rs
expression: transformation
---
import _inject from "@stylexjs/stylex/lib/stylex-inject";
var _inject2 = _inject;
import "otherFile.stylex";
import stylex from 'stylex';
import { MyTheme } from 'otherFile.stylex';
_inject2("@keyframes __hashed_var__1cb153o-B{from{color:var(--__hashed_var__1jqb1tb);}}", 1);
export const fade = "__hashed_var__1cb153o-B";
_inject2(".__hashed_var__1xwo6t1{animation-name:__hashed_var__1cb153o-B}", 3000);
"__hashed_var__1xwo6t1";